### DISCUSSION TITLE
Reset value of sibling field when a field's value changes

### DISCUSSION THREAD

**Discussion by divyamoncy** [react-querybuilder/react-querybuilder#508] [Q&A] [answered]
I have two fields named `actions` and `filter`. Both these fields are siblings. I want to reset `filter`  value when `actions` value changes. Could you please suggest a way to do this?

**Comment by jakeboone02** [comment]
Hi @divyamoncy! Could you provide a screenshot or mockup of what you want to do? I just need some clarification about what you mean by "siblings" in the context of the query builder hierarchy.

**Reply by divyamoncy** [reply]
My rule groups are of a predefined structure and always will have `action` and `filter` fields. I am using custom component for filter. What I want to do is when `action` value changes, I want to change the `filter` value just next to it.

<img width="469" alt="Screenshot 2023-05-04 at 9 00 46 AM" src="https://user-images.githubusercontent.com/22353389/236106936-81d32aab-f45b-4df7-885b-710b45cda788.png">

**Reply by divyamoncy** [reply]
This is the sample code of what I want to do.
```
const ValueEditor = (props: ValueEditorProps) => {
    const { handleOnChange } = props;
    if (field === 'filter') {
      return <Filter {...props}/>
    }
    if (field === 'action') {
      return <DefaultValueEditor {...props} handleOnChange={(value)=> { handleOnChange(value); 
      /* here I want to change the value of filter */ 
      }}/>;
    }
    return <DefaultValueEditor {...props} />
  };
```

**Reply by jakeboone02** [reply, marked as answer]
The `handleOnChange` callback can only update the `value` property of the rule it's assigned to. It can't update other rules (or even other properties of the same rule).

What you _can_ do is create a separate function that updates the entire query and call _that_ from your custom value editor instead of `handleOnChange`.

The [query tools](https://react-querybuilder.js.org/docs/utils/misc#query-tools) that RQB uses internally are exported: `add`, `update`, `remove`, and `move`. You'll want to use `update`, but send it the `path` (see [Path concepts](https://react-querybuilder.js.org/docs/tips/path)) of the sibling rule instead of the current rule.

**Reply by divyamoncy** [reply]
Thanks a lot @jakeboone02 ! :) I was able to do it by passing the query changing function in `props.context` to my Custom editor.

**Reply by divyamoncy** [reply]
Hi @jakeboone02 I'm still facing a problem, the query output is changing and filter is being reset but the UI is not updating. Could you please help?

```
const ValueEditor = (props: ValueEditorProps) => {
    const { operator, value, field, handleOnChange } = props;
    if (field === 'filter') {
        return <Filter {...props} />
    }
    if (field === 'action') {
        let pathOfFilter = [...props.path];
        pathOfFilter[pathOfFilter.length - 1] = pathOfFilter[pathOfFilter.length - 1] + 1;
        return <DefaultValueEditor {...props} handleOnChange={(value) => {
            handleOnChange(value);
            props.context.changeQuery(update(update(props.context.query, 'value', value, props.path), 'value', {}, pathOfFilter));
        }} />;
    }
    return <DefaultValueEditor {...props} />
};
```

**Reply by jakeboone02** [reply]
Would you mind setting up a CodeSandbox that I can fork? You can use one of the [examples](https://github.com/react-querybuilder/react-querybuilder/tree/main/examples) as a template.

**Reply by divyamoncy** [reply]
Hi I was able to fix it using a useEffect in filter. Thanks!
